programs/drift/src/instructions/keeper.rs

```diff
@@ -205,18 +205,26 @@ fn fill_spot_order<'a, 'b, 'c, 'info>(
 
     let (_referrer, _referrer_stats) = get_referrer_and_referrer_stats(remaining_accounts_iter)?;
 
-    let mut fulfillment_params = match fulfillment_type {
+    let mut fulfillment_params: Box<dyn SpotFulfillmentParams> = match fulfillment_type {
         Some(SpotFulfillmentType::SerumV3) => {
             let base_market = spot_market_map.get_ref(&market_index)?;
             let quote_market = spot_market_map.get_quote_spot_market()?;
-            get_serum_fulfillment_accounts(
+            Box::new(get_serum_fulfillment_params(
                 remaining_accounts_iter,
                 &ctx.accounts.state,
                 &base_market,
                 &quote_market,
-            )?
+            )?)
+        }
+        Some(SpotFulfillmentType::Match) | None => {
```

Comment:
Feels like fulfillment_type shouldn't be an option...

You should `unwrap_or(SpotFulfillmentType::Match)` upstream and plumb that through IMO

Reply:
yeah we have to leave it as an option at the ix level for backwards compatibility, but unwrap_or higher up makes sense